Compile error with BlockVectorBase<VectorType>:equ

Class `BlockVectorBase<VectorType>` and consequently all of its child classes contain the method `equ(a, U, b, V)` [here](https://github.com/dealii/dealii/blob/93c23a153520ca3715b83f06eee4331b26772427/include/deal.II/lac/block_vector_base.h#L942) and it is implemented by calling the corresponding method `VectorType.equ(a, U[i], b, V[i])`, but there is no `equ` method with 4 arguments in vector classes. There used to be such methods, but they were deprecated and removed (for example in #1021). 

Calling this `equ` method in any of the block vector classes causes compile error
```
In file included from /Applications/deal.II.app/Contents/Resources/include/deal.II/lac/generic_linear_algebra.h:21:
In file included from /Applications/deal.II.app/Contents/Resources/include/deal.II/lac/block_vector.h:25:
/Applications/deal.II.app/Contents/Resources/include/deal.II/lac/block_vector_base.h:1971:45: error: too many arguments to function call, expected 2, have 4
      components[i].equ(a, v.components[i], b, w.components[i]);
      ~~~~~~~~~~~~~~~~~                     ^~~~~~~~~~~~~~~~~~
../src/solvers/newton_iter.cc:30:26: note: in instantiation of member function 'dealii::BlockVectorBase<dealii::TrilinosWrappers::MPI::Vector>::equ' requested here
      solution.increment.equ(1., solution.current, -1., solution.previous);
                         ^
/Applications/deal.II.app/Contents/Resources/include/deal.II/lac/trilinos_vector.h:2106:5: note: 'equ' declared here
    inline void
    ^
```
How should we fix it?
1. Reintroduce `equ(a, U, b, V)` to vector classes.
2. Remove `equ(a, U, b, V)` from `BlockVectorBase`.
3. Rewrite the implementation of `BlockVectorBase<VectorType>::equ(a, U, b, V)` to use `VectorType::equ(a, U)` and `VectorType::add(b, V)`

--- a/doc/news/changes/incompatibilities/20190903RezaRastak
+++ b/doc/news/changes/incompatibilities/20190903RezaRastak
@@ -1,0 +1,3 @@
+Removed: BlockVectorBase::equ(a, U, b, V) is removed.
+<br>
+(Reza Rastak, 2019/09/03)

--- a/include/deal.II/lac/block_vector_base.h
+++ b/include/deal.II/lac/block_vector_base.h
@@ -936,15 +936,6 @@ public:
   equ(const value_type a, const BlockVector2 &V);
 
   /**
-   * U=a*V+b*W. Replacing by sum.
-   */
-  void
-  equ(const value_type       a,
-      const BlockVectorBase &V,
-      const value_type       b,
-      const BlockVectorBase &W);
-
-  /**
    * Update the ghost values by calling <code>update_ghost_values</code> for
    * each block.
    */
@@ -1944,29 +1935,6 @@ BlockVectorBase<VectorType>::scale(const BlockVector2 &v)
          ExcDimensionMismatch(n_blocks(), v.n_blocks()));
   for (size_type i = 0; i < n_blocks(); ++i)
     components[i].scale(v.block(i));
-}
-
-
-
-template <class VectorType>
-void
-BlockVectorBase<VectorType>::equ(const value_type                   a,
-                                 const BlockVectorBase<VectorType> &v,
-                                 const value_type                   b,
-                                 const BlockVectorBase<VectorType> &w)
-{
-  AssertIsFinite(a);
-  AssertIsFinite(b);
-
-  Assert(n_blocks() == v.n_blocks(),
-         ExcDimensionMismatch(n_blocks(), v.n_blocks()));
-  Assert(n_blocks() == w.n_blocks(),
-         ExcDimensionMismatch(n_blocks(), w.n_blocks()));
-
-  for (size_type i = 0; i < n_blocks(); ++i)
-    {
-      components[i].equ(a, v.components[i], b, w.components[i]);
-    }
 }
 
 

